logictest: remove 3node-tenant blocklist from some tests

asubiotto opened the pull request:
The big change here is returning nil in `crdb_internal.zones.populate`. It makes sense to me to do so in the spirit of making zone configs "optional" for phase 2 (#49445) although I'm not 100% sure this won't break anything.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50041)
<!-- Reviewable:end -->

Commit:
sql: don't return an error in crdb_internal.zones.populate if tenant

Previously, crdb_internal.zones.populate would return an error if not run
by a system tenant. Successful execution of this function is currently
required for statements like SHOW CREATE TABLE when run by a tenant.

This commit still avoids populating crdb_internal.zones but returns no error,
which is equivalent to an empty crdb_internal.zones table. Since tenants will
not have specific zone configs for phase 2, this is fine.

This commit also removes the 3node-tenant blocklist directive from tests that
now pass.

Release note: None (multitenancy work)

Commit:
logictest: remove 3node-tenant blocklist from some tests

Release note: None (testing change)

asubiotto commented:
CI failure is tracked in #50000

asubiotto commented:
bors r=tbg
